### PR TITLE
Allow queries with keys that are intentionally empty

### PR DESCRIPTION
**Pull request by doaa-altarawy**
## Description
Fixes issue #157 

## Questions
1. In results, only basis and keywords are optional, thus allowed to be None, other required fields can't be empty so there is no meaning to search for, say, program=None; it doesn't exist.
1. In molecule, it doesn't support returing the full list (i.e., get_molecule() -> all molecules up to the max_limit).
1. Collection has both fields (collection, name) as required, so no meaning to search for name=None. Also, the Web hander gets any empty query to None in Pydantic validation. Therefore, it assumes storage.get_collection(name=None, collection=None) to return ALL collections.

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by doaa-altarawy** [comment]
This is ready

File: qcfractal/storage_sockets/mongoengine_socket.py [hunk 1]
```diff
@@ -1041,18 +1042,18 @@ def get_results(self,
...
+            query['basis'] = basis if basis != 'null' else None
+        if molecule is not None:
             query['molecule'], _ = _str_to_indices_with_errors(molecule)
-        if driver:
+        if driver is not None:
             query['driver'] = driver
-        if keywords:
-            query['keywords'] = keywords
+        if keywords is not None:
+            query['keywords'] = keywords if keywords != 'null' else None
```
Comment (dgasmith) [inline qcfractal/storage_sockets/mongoengine_socket.py]: I think this should be `keywords if keywords != '' else None` now?

Comment (doaa-altarawy) [reply qcfractal/storage_sockets/mongoengine_socket.py]: Yes of course

Comment (doaa-altarawy) [reply qcfractal/storage_sockets/mongoengine_socket.py]: I think we should remove this now since we already have the default: `""`

Comment (dgasmith) [reply qcfractal/storage_sockets/mongoengine_socket.py]: +1 for that.

**Comment by dgasmith** [comment]
It seems `None` is fairly special to Mongo so we should keep these fields as None in the database. At the moment we are fixing this in the REST query. Which is not ideal. 

However, this fixes the core issue and keeps DB entires as None which I think is the primary thing today. I vote merging this in and opening an issue to think about this a bit more.

**Pull request merged**